server/task.go

```diff
@@ -626,8 +604,8 @@ func (s *Server) changeTaskStatusWithPatch(ctx context.Context, task *api.Task,
 		}
 		// Set database labels, except bb.environment is immutable and must match instance environment.
 		// This needs to be after we compose database relationship.
-		if err == nil && databaseCreate.Labels != nil && *databaseCreate.Labels != "" {
-			if err := s.setDatabaseLabels(ctx, *databaseCreate.Labels, database, project, databaseCreate.CreatorID, false /* validateOnly */); err != nil {
+		if err == nil && database.Labels != "" {
```

Comment:
We should also move this logic to the executor which is native to database creation.